### ISSUE TITLE
Noisy crash on server timeout

### ISSUE THREAD

**Issue by michalmuskala** [hexpm/hex#498]
There were some intermittent issues with the server today and the crash was pretty nasty. It should be possible to fail gracefully in this situation.

```
Failed to check for new Hex version
:timeout
Failed to fetch record for 'hexpm/cowlib' from registry (using cache)
** (exit) exited in: GenServer.call(Hex.Registry.Server, {:versions, "hexpm", "connection"}, 60000)
    ** (EXIT) an exception was raised:
        ** (FunctionClauseError) no function clause matching in Hex.Registry.Server.missing_status?/1
            (hex) lib/hex/registry/server.ex:365: Hex.Registry.Server.missing_status?({:error, :timeout})
            (hex) lib/hex/registry/server.ex:355: Hex.Registry.Server.print_error/4
            (hex) lib/hex/registry/server.ex:343: Hex.Registry.Server.write_result/4
            (hex) lib/hex/registry/server.ex:196: Hex.Registry.Server.handle_info/2
            (stdlib) gen_server.erl:616: :gen_server.try_dispatch/4
            (stdlib) gen_server.erl:686: :gen_server.handle_msg/6
            (stdlib) proc_lib.erl:247: :proc_lib.init_p_do_apply/3
    (elixir) lib/gen_server.ex:831: GenServer.call/3
    (hex) lib/hex/remote_converger.ex:261: Hex.RemoteConverger.check_dep/3
    (elixir) lib/enum.ex:740: anonymous fn/3 in Enum.each/2
    (stdlib) lists.erl:1263: :lists.foldl/3
    (elixir) lib/enum.ex:1907: Enum.each/2
    (hex) lib/hex/remote_converger.ex:39: Hex.RemoteConverger.converge/2
    (mix) lib/mix/dep/converger.ex:95: Mix.Dep.Converger.all/4
    (mix) lib/mix/dep/converger.ex:51: Mix.Dep.Converger.converge/4

12:38:10.611 [error] GenServer Hex.Registry.Server terminating
** (FunctionClauseError) no function clause matching in Hex.Registry.Server.missing_status?/1
    (hex) lib/hex/registry/server.ex:365: Hex.Registry.Server.missing_status?({:error, :timeout})
    (hex) lib/hex/registry/server.ex:355: Hex.Registry.Server.print_error/4
    (hex) lib/hex/registry/server.ex:343: Hex.Registry.Server.write_result/4
    (hex) lib/hex/registry/server.ex:196: Hex.Registry.Server.handle_info/2
    (stdlib) gen_server.erl:616: :gen_server.try_dispatch/4
    (stdlib) gen_server.erl:686: :gen_server.handle_msg/6
    (stdlib) proc_lib.erl:247: :proc_lib.init_p_do_apply/3
Last message: {:get_package, "hexpm", "cowlib", {:error, :timeout}}
State: %{closing_fun: nil, ets: #Reference<0.76765314.3236036614.102474>, fetched: #MapSet<[]>, offline?: false, path: "/Users/michalmuskala/.hex/cache.ets", pending: #MapSet<[{"hexpm", "connection"}, {"hexpm", "cowboy"}, {"hexpm", "cowlib"}, {"hexpm", "db_connection"}, {"hexpm", "decimal"}, {"hexpm", "ecto"}, {"hexpm", "gettext"}, {"hexpm", "mime"}, {"hexpm", "phoenix"}, {"hexpm", "phoenix_ecto"}, {"hexpm", "phoenix_pubsub"}, {"hexpm", "plug"}, {"hexpm", "poison"}, {"hexpm", "poolboy"}, {"hexpm", "postgrex"}, {"hexpm", "ranch"}, {"hexpm", "uuid"}]>, waiting: %{{"hexpm", "connection"} => [{{#PID<0.73.0>, #Reference<0.76765314.3235905537.101454>}, #Function<3.75490635/0 in Hex.Registry.Server.handle_call/3>}]}}
```
and
```
Failed to check for new Hex version
:timeout
Failed to fetch record for 'hexpm/cowlib' from registry (using cache)
** (exit) exited in: GenServer.call(Hex.Registry.Server, {:versions, "hexpm", "connection"}, 60000)
    ** (EXIT) an exception was raised:
        ** (FunctionClauseError) no function clause matching in Hex.Registry.Server.missing_status?/1
            (hex) lib/hex/registry/server.ex:365: Hex.Registry.Server.missing_status?({:error, :timeout})
            (hex) lib/hex/registry/server.ex:355: Hex.Registry.Server.print_error/4
            (hex) lib/hex/registry/server.ex:343: Hex.Registry.Server.write_result/4
            (hex) lib/hex/registry/server.ex:196: Hex.Registry.Server.handle_info/2
            (stdlib) gen_server.erl:616: :gen_server.try_dispatch/4
            (stdlib) gen_server.erl:686: :gen_server.handle_msg/6
            (stdlib) proc_lib.erl:247: :proc_lib.init_p_do_apply/3
    (elixir) lib/gen_server.ex:831: GenServer.call/3
    (hex) lib/hex/remote_converger.ex:261: Hex.RemoteConverger.check_dep/3
    (elixir) lib/enum.ex:740: anonymous fn/3 in Enum.each/2
    (stdlib) lists.erl:1263: :lists.foldl/3
    (elixir) lib/enum.ex:1907: Enum.each/2
    (hex) lib/hex/remote_converger.ex:39: Hex.RemoteConverger.converge/2
    (mix) lib/mix/dep/converger.ex:95: Mix.Dep.Converger.all/4
    (mix) lib/mix/dep/converger.ex:51: Mix.Dep.Converger.converge/4

12:37:34.795 [error] GenServer Hex.Registry.Server terminating
** (FunctionClauseError) no function clause matching in Hex.Registry.Server.missing_status?/1
    (hex) lib/hex/registry/server.ex:365: Hex.Registry.Server.missing_status?({:error, :timeout})
    (hex) lib/hex/registry/server.ex:355: Hex.Registry.Server.print_error/4
    (hex) lib/hex/registry/server.ex:343: Hex.Registry.Server.write_result/4
    (hex) lib/hex/registry/server.ex:196: Hex.Registry.Server.handle_info/2
    (stdlib) gen_server.erl:616: :gen_server.try_dispatch/4
    (stdlib) gen_server.erl:686: :gen_server.handle_msg/6
    (stdlib) proc_lib.erl:247: :proc_lib.init_p_do_apply/3
Last message: {:get_package, "hexpm", "cowlib", {:error, :timeout}}
State: %{closing_fun: nil, ets: #Reference<0.4101339474.2162294792.111824>, fetched: #MapSet<[]>, offline?: false, path: "/Users/michalmuskala/.hex/cache.ets", pending: #MapSet<[{"hexpm", "connection"}, {"hexpm", "cowboy"}, {"hexpm", "cowlib"}, {"hexpm", "db_connection"}, {"hexpm", "decimal"}, {"hexpm", "ecto"}, {"hexpm", "gettext"}, {"hexpm", "mime"}, {"hexpm", "phoenix"}, {"hexpm", "phoenix_ecto"}, {"hexpm", "phoenix_pubsub"}, {"hexpm", "plug"}, {"hexpm", "poison"}, {"hexpm", "poolboy"}, {"hexpm", "postgrex"}, {"hexpm", "ranch"}, {"hexpm", "uuid"}]>, waiting: %{{"hexpm", "connection"} => [{{#PID<0.73.0>, #Reference<0.4101339474.2162163719.112039>}, #Function<3.75490635/0 in Hex.Registry.Server.handle_call/3>}]}}

12:37:34.796 [error] GenServer Hex.UpdateChecker terminating
** (stop) exited in: GenServer.call(Hex.Registry.Server, {:last_update, {{2018, 1, 31}, {11, 37, 34}}}, 60000)
    ** (EXIT) an exception was raised:
        ** (FunctionClauseError) no function clause matching in Hex.Registry.Server.missing_status?/1
            (hex) lib/hex/registry/server.ex:365: Hex.Registry.Server.missing_status?({:error, :timeout})
            (hex) lib/hex/registry/server.ex:355: Hex.Registry.Server.print_error/4
            (hex) lib/hex/registry/server.ex:343: Hex.Registry.Server.write_result/4
            (hex) lib/hex/registry/server.ex:196: Hex.Registry.Server.handle_info/2
            (stdlib) gen_server.erl:616: :gen_server.try_dispatch/4
            (stdlib) gen_server.erl:686: :gen_server.handle_msg/6
            (stdlib) proc_lib.erl:247: :proc_lib.init_p_do_apply/3
    (elixir) lib/gen_server.ex:831: GenServer.call/3
    (hex) lib/hex/update_checker.ex:62: Hex.UpdateChecker.handle_info/2
    (stdlib) gen_server.erl:616: :gen_server.try_dispatch/4
    (stdlib) gen_server.erl:686: :gen_server.handle_msg/6
    (stdlib) proc_lib.erl:247: :proc_lib.init_p_do_apply/3
Last message: {#Reference<0.4101339474.2162163719.111525>, {:installs, {:error, :timeout}}}
```

**Comment by ericmj** [comment]
What hex version are you using? (`mix hex.info`)

**Comment by michalmuskala** [comment]
```
Hex:    0.17.1
Elixir: 1.6.0-dev
OTP:    20.1

Built with: Elixir 1.5.1 and OTP 18.3
```

**Comment by ericmj** [comment]
It should be fixed on 0.17.3. It will print something like this:

```
Failed to fetch record for 'hexpm/ex_machina' from registry (using cache)
:timeout
Failed to fetch record for 'hexpm/phoenix' from registry (using cache)
:timeout
Failed to fetch record for 'hexpm/postgrex' from registry (using cache)
:timeout
```

**Comment by michalmuskala** [comment]
Awesome. I'm sorry for creating the noise on this.
